tt_fabric/mesh_graph.hpp

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "magic_enum.hpp"
+#include "tt_metal/common/assert.hpp"
+#include "tt_metal/llrt/tt_cluster.hpp"
+#include "tt_metal/third_party/umd/device/tt_arch_types.h"
+
+namespace tt::tt_fabric {
+struct Chip {
+    tt::ARCH arch;
+    std::uint32_t num_eth_ports_per_direction;
```

Comment:
Initialize all of these to either 0 or `Invalid` to avoid accidental UB.

`Chip` also sounds overly generic, perhaps `ChipSpec` would be a better name?